spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -0,0 +1,149 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.WritableResource;
+import org.springframework.util.Assert;
+
+/**
+ * Implements {@link WritableResource} for buckets in Google Cloud Storage.
+ */
+public class GoogleStorageResourceBucket implements WritableResource {
+
+	private final Storage storage;
+
+	private final String bucketName;
+
+	public GoogleStorageResourceBucket(Storage storage, String bucketName,
+			boolean createBucketIfNotExists) {
+		Assert.notNull(storage, "Storage object can not be null");
+		Assert.notNull(bucketName, "Bucket name can not be null");
+
+		this.storage = storage;
+		this.bucketName = bucketName;
+
+		createBucketIfRequired(createBucketIfNotExists);
+	}
+
+	private void createBucketIfRequired(boolean createBucketIfNotExists) {
+		if (!exists() && createBucketIfNotExists) {
+			this.storage.create(BucketInfo.newBuilder(this.bucketName).build());
+		}
+	}
+
+	@Override
+	public boolean exists() {
+		return resolve() != null;
+	}
+
+	@Override
+	public boolean isReadable() {
+		return false;
+	}
+
+	@Override
+	public boolean isOpen() {
+		return false;
+	}
+
+	@Override
+	public URL getURL() throws IOException {
+		return getURI().toURL();
+	}
+
+	@Override
+	public URI getURI() throws IOException {
+		URI uri = null;
+		try {
+			uri = new URI(GoogleStorageProtocolResolver.PROTOCOL + getFilename());
+		}
+		catch (URISyntaxException e) {
+			throw new IOException("Invalid URI syntax", e);
+		}
+		return uri;
+	}
+
+	@Override
+	public File getFile() {
+		throw new UnsupportedOperationException(
+				"GCS bucket is not in the local filesystem.");
+	}
+
+	@Override
+	public long contentLength() {
+		throw new UnsupportedOperationException(
+				"Cannot get content length for a storage bucket.");
+	}
+
+	@Override
+	public long lastModified() {
+		throw new UnsupportedOperationException(
+				"Cannot get last-modified for a storage bucket.");
+	}
+
+	@Override
+	public Resource createRelative(String relativePath) throws IOException {
+		return new GoogleStorageResourceObject(this.storage,
+				getURI().toString() + (relativePath.startsWith("/") ? "" : "/")
+						+ relativePath);
+	}
+
+	@Override
+	public String getFilename() {
+		Bucket bucket = resolve();
```

Comment:
You already have `this.bucketName`. Why do `resolve()`?

Reply:
Yea I'll make the change. Originally I had wanted it to return the name only if it exists, but that's not necessary.